sherpa-onnx/csrc/provider-config.cc

```diff
@@ -0,0 +1,158 @@
+  po->Register("trt-max-workspace-size", &trt_max_workspace_size,
+              "Set TensorRT EP GPU memory usage limit.");
+  po->Register("trt-max-partition-iterations", &trt_max_partition_iterations,
+              "Limit partitioning iterations for model conversion.");
+  po->Register("trt-min-subgraph-size", &trt_min_subgraph_size,
+              "Set minimum size for subgraphs in partitioning.");
+  po->Register("trt-fp16-enable", &trt_fp16_enable,
+              "Enable FP16 precision for faster performance.");
+  po->Register("trt-detailed-build-log", &trt_detailed_build_log,
+              "Enable detailed logging of build steps.");
+  po->Register("trt-engine-cache-enable", &trt_engine_cache_enable,
+              "Enable caching of TensorRT engines.");
+  po->Register("trt-engine-cache-path", &trt_engine_cache_path,
+              "Set path to store cached TensorRT engines.");
+  po->Register("trt-timing-cache-enable", &trt_timing_cache_enable,
+              "Enable use of timing cache to speed up builds.");
+  po->Register("trt-timing-cache-path", &trt_timing_cache_path,
+              "Set path for storing timing cache.");
+  po->Register("trt-dump-subgraphs", &trt_dump_subgraphs,
+              "Dump optimized subgraphs for debugging.");
+}
+
+bool TensorrtConfig::Validate() const {
+  if (trt_max_workspace_size < 0) {
+    SHERPA_ONNX_LOGE("trt_max_workspace_size: '%u' is not valid.",
+        trt_max_workspace_size);
+    return false;
+  }
+  if (trt_max_partition_iterations < 0) {
+    SHERPA_ONNX_LOGE("trt_max_partition_iterations: '%u' is not valid.",
+        trt_max_partition_iterations);
+    return false;
+  }
+  if (trt_min_subgraph_size < 0) {
+    SHERPA_ONNX_LOGE("trt_min_subgraph_size: '%u' is not valid.",
+        trt_min_subgraph_size);
+    return false;
+  }
+  if (trt_fp16_enable != true || trt_fp16_enable != false) {
+    SHERPA_ONNX_LOGE("trt_fp16_enable: '%d' is not valid.", trt_fp16_enable);
+    return false;
+  }
+  if (trt_detailed_build_log != true || trt_detailed_build_log != false) {
+    SHERPA_ONNX_LOGE("trt_detailed_build_log: '%d' is not valid.",
+        trt_detailed_build_log);
+    return false;
+  }
+  if (trt_engine_cache_enable != true || trt_engine_cache_enable != false) {
+    SHERPA_ONNX_LOGE("trt_engine_cache_enable: '%d' is not valid.",
+        trt_engine_cache_enable);
+    return false;
+  }
+  if (trt_timing_cache_enable != true || trt_timing_cache_enable != false) {
+    SHERPA_ONNX_LOGE("trt_timing_cache_enable: '%d' is not valid.",
+        trt_timing_cache_enable);
+    return false;
+  }
+  if (trt_dump_subgraphs != true || trt_dump_subgraphs != false) {
+    SHERPA_ONNX_LOGE("trt_dump_subgraphs: '%d' is not valid.",
+        trt_dump_subgraphs);
+    return false;
+  }
+
+  return true;
+}
+
+std::string TensorrtConfig::ToString() const {
+  std::ostringstream os;
+
+  os << "TensorrtConfig(";
+  os << "trt_max_workspace_size=\"" << trt_max_workspace_size << "\", ";
```

Comment:
```suggestion
  os << "trt_max_workspace_size=" << trt_max_workspace_size << ", ";
```

You don't need to use `""` to enclose an integer. Please fix similar issues below.